test/powershell/engine/ExecutionPolicy.Tests.ps1

```diff
@@ -851,6 +851,102 @@ ZoneId=$FileType
     }
 }
 
+Describe "Validate Set-ExecutionPolicy -Scope" -Tags "CI" {
+
+    BeforeAll {
+        $originalPolicies = Get-ExecutionPolicy -list
+
+        # LocalMachine Execution Policy Can only be set by Administrators
+        $IsAdministrator = $false
+
+        $windowsIdentity = [Security.Principal.WindowsIdentity]::GetCurrent()
+        $windowsPrincipal = new-object 'Security.Principal.WindowsPrincipal' $windowsIdentity
+        if ($windowsPrincipal.IsInRole("Administrators") -eq $true) {
+            $IsAdministrator = $true
+        }
+
+
+        # Calls Set-ExecutionPolicy with a known-bad Scope and expects failure.
+        # It is defined here so that it will be available at It scope.
+        function VerfiyBlockedSetExecutionPolicy
+        {
+            param(
+                [string]
+                $policyScope
+            )
+            $fqeid = ""
+            try {
+                Set-ExecutionPolicy -Scope $policyScope -ExecutionPolicy Restricted
+            }
+            catch {
+                $fqeid = $_.FullyQualifiedErrorId
+            }
+
+            $fqeid | Should Be "CantSetGroupPolicy,Microsoft.PowerShell.Commands.SetExecutionPolicyCommand"
+        }
+    }
+
+    AfterAll {
+        foreach ($scopedPolicy in $originalPolicies)
+        {
+            if (($scopedPolicy.Scope -eq "Process") -or
+                ($scopedPolicy.Scope -eq "CurrentUser"))
+            {
+                try {
+                    Set-ExecutionPolicy -Scope $scopedPolicy.Scope -ExecutionPolicy $scopedPolicy.ExecutionPolicy -Force
+                }
+                catch {
+                    if ($_.FullyQualifiedErrorId -ne "ExecutionPolicyOverride,Microsoft.PowerShell.Commands.SetExecutionPolicyCommand")
+                    {
+                        # Re-throw unrecognized exceptions. Otherwise, swallow the exception that warns about overridden policies
+                        throw $_
+                    }
+                }
+            }
+            elseif(($scopedPolicy.Scope -eq "LocalMachine") -and
+                    $IsAdministrator)
```

Comment:
I assume that you meant LocalMachine. MachinePolicy is not writable via the cmdlets.
The LocalMachine setting is stored within $PSHOME and shares the same ownership as the entity that created it. The Windows installer runs as Administrator, so the file requires Administrator access during normal runtime. I wrote the test for the more restrictive scenario.

An alternative is to run the test always and have it fail when run by regular user in the situation where it requires Administrator creds. There are 5 or so tests that behave that way.

What is your preference?